backend/src/database.ts

```diff
@@ -169,19 +169,19 @@ export function createNewPseudoPlayer(eloId: string): dbPseudoPlayer {
 
 // Adds a puzzle to the list of the players' solved puzzles, if it is not
 // already marked as solved.
-export async function addPseudoPlayerSolvedPuzzle(
-  eloId: string,
+export async function addPlayerSolvedPuzzle(
+  idToken: string,
   name: string,
   puzzleId: string
 ): Promise<void> {
-  if (!connectedToDB) return;
-  let p = await PseudoPlayer.findOne({ eloId: eloId });
-  // If the pseudoplayer is not in the DB yet because they have not played any game yet, create it
+  if (!connectedToDB || isGuest(idToken)) return;
+  let p = await Player.findOne({ idToken: idToken });
+  // If the player is not in the DB yet because they have not played any game yet, create it
   // so we can keep track of their puzzle completions.
   if (!p) {
     const timestamp = Date.now();
-    console.log("creating new pseudoPlayer");
-    p = new PseudoPlayer(createNewPseudoPlayer(eloId));
+    console.log("creating new Player");
+    p = new Player(createNewPlayer(""));
```

Comment:
why are you passing an empty string here, instead of the idToken?

Reply:
Good catch!

Reply:
fixed!